module/os/linux/zfs/vdev_disk.c

```diff
@@ -230,7 +230,7 @@ vdev_elevator_switch(vdev_t *v, char *elevator)
 
 static int
 vdev_disk_open(vdev_t *v, uint64_t *psize, uint64_t *max_psize,
-    uint64_t *ashift)
+    uint64_t *ashift, uint64_t *pshift)
```

Comment:
After taking a look at this I wonder if we actually want to add this complexity.  I'd forgotten this, but years ago we updated `vdev_disk_open()` on Linux to always return the physical sector size for the ashift when available instead of the logical size.  Would it be reasonable to do the same thing on FreeBSD?

Reply:
mav said he'd follow up tomorrow, but today he said:
```
@mmacy I need to reread the patch, I was a while ago and mostly by avg@ IIRC. But my main worry about his proposition to use pshift as ashift is that ZFS may not be able to read pool created with small ashift if after creation OS decided to report some bigger pshift. IIRC the logic of passing separately is to differentiate creation stage from import stage (the second should be more relaxed).
```